### ISSUE TITLE
Feature request: show progress and remaining ride length during live ride.

### ISSUE THREAD

**Issue by bdsl** [cyclestreets/android#95]
Could be distance and/or estimated time or effort.

**Comment by HilaryN** [comment]
I'm just thinking about where to put this on the screen and my thoughts so far are to reduce the size of the text of the current speed and to use that box (bottom left) to include remaining distance etc.
<img width="218" alt="Current speed" src="https://user-images.githubusercontent.com/59799824/80305136-f0141d80-87b2-11ea-9b7c-94ee27c66d9d.PNG">

**Comment by HilaryN** [comment]
Like this (these are hard-coded values for now):
![image](https://user-images.githubusercontent.com/59799824/80615094-2bbf1980-8a37-11ea-91bf-78c8eda7a048.png)

**Comment by HilaryN** [comment]
OK, I'll have a go at producing a screenshot of that.  Small text like in these 2 screenshots, or large like the original speed?  (Or medium? :-) )  I'll do some examples of each so you can see.

**Comment by HilaryN** [comment]
In A I've moved the speed to the top:
<img width="250" alt="Screenshot A" src="https://user-images.githubusercontent.com/59799824/80650772-82931600-8a6c-11ea-95ec-9658e5adaf3b.PNG">
Here it is all on one line (needs to have an offset on the right, though):
<img width="599" alt="Screenshot B" src="https://user-images.githubusercontent.com/59799824/80650779-8757ca00-8a6c-11ea-9a54-1b647c4a02bf.PNG">
This is "medium" text - too large:
<img width="599" alt="Screenshot C" src="https://user-images.githubusercontent.com/59799824/80650728-714a0980-8a6c-11ea-98d0-aa013d3ff612.PNG">
I think this is my favourite.  The boxes could be split out, maybe, and the copyright would need moving up above the green box (as it would in B):
<img width="600" alt="Screenshot D" src="https://user-images.githubusercontent.com/59799824/80650750-7ad37180-8a6c-11ea-90a2-8372af80d7db.PNG">

**Comment by mvl22** [comment]
Personally I most prefer D!

**Comment by jezhiggins** [comment]
I'd suggest strapping your phone on to your bars and going for a ride around before settling on something. That's primarily how I arrived at the relatively large text for your speed in the first place. It's got to be glanceable as you ride, because otherwise anything on screen is, unfortunately, pretty useless.

**Comment by HilaryN** [comment]
Yes, I was worried about that, too, (especially with all the potholes and rough surfaces on the roads which make the bike shake!) and certainly will - this is just to get some initial feedback.  Noted about the speed - I'll put that back to large.  I tend to ignore it as it's always way faster than my actual speed.
  Not so much of an issue with the remaining journey as one can easily check that when paused, but maybe could do with being bigger.  Or configurable.

**Comment by HilaryN** [comment]
(Copyright still to be shifted up.)  Thoughts / suggestions?
<img width="526" alt="Screenshot E" src="https://user-images.githubusercontent.com/59799824/81748832-68acf680-94a2-11ea-9341-bd51cfcfad65.PNG">

**Comment by mvl22** [comment]
Looks good! Great improvement.

For simplicity of implementation, to avoid having to special-case moving the attribution about for different screens, I would just increase the box size by a few pixels, make the right-hand side font a few pixels smaller, then shift all the text up slightly. Only needs slight adjustments to keep the attribution where it is, in my view.

**Comment by oliverlockwood** [comment]
I agree this layout looks generally reasonable. @jezhiggins any thoughts from you?

However I disagree with @mvl22's suggestion regarding the attribution. Its location is already special cased in LiveRide, to be to centred on the map that remains to the right of the original speed-only box. So it would be better to change the special casing to elevate it instead of shifting right, and shouldn't be that hard.

@HilaryN please be aware that with the recent AndroidX changes that have gone in, you will need to make sure you merge from master into your branch, and ensure any view/layout changes you've made also reference the correct classes in AndroidX, not the support libraries.

**Comment by HilaryN** [comment]
Yes, I've seen the code is already there to shift it (and to leave it at the bottom I would probably have to shift it under the speed anyway), so just needs changing slightly.
Yes, I saw the changes, in particular the map rotation.  I've still got lots more testing to do, which I'm saving til the end - I'll merge in the latest changes before I do that.

@jezhiggins if you want to take it out on a ride, let me know and I'll upload the code to my fork.

I'm wondering if I can squeeze a small text "Remaining journey" above.  The box is starting to look quite thick, though (and here in F I've made the remaining dist etc a bit smaller: offset * 1.8 rather than offset*2 in E).  These screenshots are on Pixel 3 (emulator), which seems to squeeze the text closer than on Pixel 2, for example.

<img width="517" alt="Screenshot F" src="https://user-images.githubusercontent.com/59799824/81797613-ef93ba80-9506-11ea-87a1-8395dca3fc0c.PNG">

**Comment by HilaryN** [comment]
Another possibility would be to have either remaining time OR ETA (set as a choice in the preferences), to make the Remaining journey part narrower.

**Comment by oliverlockwood** [comment]
I'd be keen on that suggestion. Then you could fit remaining distance and (time or ETA) onto one line, perhaps?

**Comment by oliverlockwood** [comment]
Maybe ETA needs that written next to it also, to make it obvious what it is

**Comment by mvl22** [comment]
Looks good, but I think "journey" is unnecessary - "Remaining:" is sufficient, as people know it's a journey already.

**Comment by mvl22** [comment]
> remaining time OR ETA

Personally I think remaining time is better than ETA.  People might decide to stop to catch a breath, or go to a shop on the way, or whatever.

**Comment by HilaryN** [comment]
I prefer ETA as it saves having try to calculate, and more importantly I can't see the phone time when cycling as it's far too small, and my watch is usually covered by my sleeve... :-)  (Btw ETA does continually update at the same time as remaining distance / time.)

**Comment by HilaryN** [comment]
> I'd be keen on that suggestion. Then you could fit remaining distance and (time or ETA) onto one line, perhaps?

They won't fit on one line, unfortunately :-(  But the two lines of mid-size text only take a tiny bit more height than the large speed text.

I've made a change to only show the text ("Remaining  / ETA") for the first segment.  By then one hopefully knows what they are, so no need to display the text (which is too small to read anyway when cycling) and the box height can then be reduced. (Attribution moved accordingly.)  So best of both worlds: text explaining what the numbers are at the beginning, but not taking up screen space for most of the journey.

<img width="514" alt="Screenshot G" src="https://user-images.githubusercontent.com/59799824/82152324-3dd2f180-9858-11ea-99c7-6dfd04424fe9.PNG">

I'm working on the preferences as well.

**Comment by HilaryN** [comment]
I was going to suggest not having the preferences for time/ETA as it all seemed to fit OK with slightly smaller text.  However, I tried on the Nexus One emulator today and it doesn't fit :-( 
Would it be OK if I made the km/h small text?  Maybe conditional, so only when the the text is overlapping.

<img width="483" alt="Nexus One" src="https://user-images.githubusercontent.com/59799824/83302697-b42a0900-a1f3-11ea-92f8-9f35c0b89222.PNG">

**Comment by HilaryN** [comment]
I tried adding am/pm (if phone setting is 12-hour) but it takes up far too much space.  Even when I put it in small text, it's still taking up more space than I would like.  What do you think - do we definitely need it?
Left is Pixel 3 emulator (US locale), right is my phone, Fairphone 2:

<img width="450" alt="ampm midsize" src="https://user-images.githubusercontent.com/59799824/83534201-775a5c80-a4e8-11ea-83c1-60b76a84bd72.PNG">
<img width="443" alt="ampm smallsize" src="https://user-images.githubusercontent.com/59799824/83534235-7e816a80-a4e8-11ea-9c4b-4f613dd79185.PNG">
